crates/stc_ts_file_analyzer/src/analyzer/scope/vars.rs

```diff
@@ -1043,6 +1032,36 @@ impl Analyzer<'_, '_> {
         })
         .context("tried to ensure iterator")
     }
+
+    fn regist_destructure(&mut self, span: Span, ty: Option<&Type>) -> u32 {
+        match ty {
```

Comment:
Please use `ty.as_ref().map(Type::normalize)` instead

Reply:
Are you talking about this?
```rs
    fn regist_destructure(&mut self, span: Span, ty: Option<&Type>) -> u32 {
        match ty.as_ref().map(Type::normalize) {
```

Reply:
Yes